core/trino-main/src/main/java/io/trino/server/ui/OAuth2WebUiAuthenticationFilter.java

```diff
@@ -163,9 +163,12 @@ private void redirectForNewToken(ContainerRequestContext request, String refresh
     {
         OAuth2Client.Response response = client.refreshTokens(refreshToken);
         String serializedToken = tokenPairSerializer.serialize(TokenPair.fromOAuth2Response(response));
-        request.abortWith(Response.temporaryRedirect(request.getUriInfo().getRequestUri())
-                .cookie(OAuthWebUiCookie.create(serializedToken, tokenExpiration.map(expiration -> Instant.now().plus(expiration)).orElse(response.getExpiration())))
-                .build());
+        Response.ResponseBuilder builder = Response.temporaryRedirect(request.getUriInfo().getRequestUri())
+                .cookie(OAuthWebUiCookie.create(serializedToken, tokenExpiration.map(expiration -> Instant.now().plus(expiration)).orElse(response.getExpiration())));
+        if (response.getIdToken().isPresent()) {
```

Comment:
I don't think we'll get a new ID token on token refresh. I think we receive ID token only once, when the initial authentication succeeds, after that we get only access tokens.